### PR TITLE
Fix type mismatch GraphQL

### LINKED ISSUE #5644: Type mismatch GraphQL - Typescript
While working on #5476 I found a couple of places where the usage in Typescript did not match the GraphQL definition, i.e. arguments were optional in GraphQL but not treated as such in code.

## Acceptance criteria

*   [ ] Add types back at the following files and adapt type usage or GraphQL definition to match
    *   [ ] https://github.com/ParabolInc/parabol/blob/33c3478b8699f0a61e092b533c36e4e6ac03c704/packages/server/graphql/mutations/addTeam.ts#L35
    *   [ ] https://github.com/ParabolInc/parabol/blob/33c3478b8699f0a61e092b533c36e4e6ac03c704/packages/server/graphql/mutations/createReflection.ts#L32
    *   [ ] https://github.com/ParabolInc/parabol/blob/33c3478b8699f0a61e092b533c36e4e6ac03c704/packages/server/graphql/mutations/updateTaskDueDate.ts#L28
    *   [ ] https://github.com/ParabolInc/parabol/blob/33c3478b8699f0a61e092b533c36e4e6ac03c704/packages/server/graphql/mutations/updateTeamName.ts#L24

**Estimated effort:** 4 hours

### PR DESCRIPTION
**Pull request by tianrunhe**
Fixed #5644

### CODE REVIEW

File: packages/server/graphql/mutations/updateTeamName.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ export default {
   },
   async resolve(
     _source: unknown,
-    {updatedTeam}, //FIXME type mismatch: {updatedTeam: UpdatedTeamInputType},
+    args: {updatedTeam: UpdatedTeamInputType},
```
Comment (Dschoordsch) [inline packages/server/graphql/mutations/updateTeamName.ts:24]: +1 we could still destructure here

**Review by Dschoordsch — CHANGES_REQUESTED** [review]

**Comment by mattkrick** [comment]
This is caused by a bug in GraphQL. 
It was fixed in v16 https://github.com/graphql/graphql-js/pull/3328
We're still pinned on v15 because our version of relay is old.
I recommend the following:
- Check to see if the latest version of relay supports graphql v16
- bump to the latest version of relay & latest version of graphql in a separate PR
- come back to this PR & amend it with the fixed typings

**Comment by tianrunhe** [comment]
> This is caused by a bug in GraphQL. It was fixed in v16 [graphql/graphql-js#3328](https://github.com/graphql/graphql-js/pull/3328) We're still pinned on v15 because our version of relay is old. I recommend the following:
> 
> * Check to see if the latest version of relay supports graphql v16
> * bump to the latest version of relay & latest version of graphql in a separate PR
> * come back to this PR & amend it with the fixed typings

Looks like the latest relay v13.1.1 supports to [GraphQL v15.3.0](https://github.com/facebook/relay/blob/v13.1.1/package.json#L55)

**Review by Dschoordsch — APPROVED** [review]

**Comment by Dschoordsch** [comment]
Did some smoke tests (creating team, creating reflection, updating team name and add task due date) and still works.

**Comment by mattkrick** [comment]
ah such a bummer about relay!
Looks like it was working in their RC https://github.com/facebook/relay/issues/3705, but maybe it didn't land in the official release?

**Comment by mattkrick** [comment]
LGTM

**Pull request merged**